src/wallet/wallet.cpp

```diff
@@ -1731,7 +1731,8 @@ CAmount CWallet::GetColdStakingBalance() const
 
 CAmount CWallet::GetStakingBalance(const bool fIncludeColdStaking) const
 {
-    return GetBalance() + (fIncludeColdStaking ? GetColdStakingBalance() : 0);
+    const CAmount& nBalance = GetBalance() - GetDelegatedBalance() - GetLockedCoins();
+    return nBalance + (fIncludeColdStaking ? GetColdStakingBalance() : 0);
```

Comment:
Each of the `GetBalanceXXX` methods call are going through all of the txs inside the wallet.

Would be better to have a single method filtering the transactions that needs to be filter only for this (one single loop over all of the txs and not three).

Reply:
absolutely, nice idea.